XavierGuertin/WildfireVisualizationProject
# #53 - feat(UI): Added loading bar component to be used for any loading processes

### Summary

This PR introduces the following changes:

- Added a `LoadingOverlay` component with a loading bar that indicates the dataset loading progress.
- Configured the `LoadingOverlay` component to dim the background and prevent user interactions with other elements while loading.
- Integrated the `LoadingOverlay` component within the `Layout` component to allow it to be triggered when the dataset is loading.

---

### Changes Made

- **Created `LoadingOverlay.tsx` in `components/`**:
  - Displays a full-screen overlay with a loading bar and percentage indicator.
  - Progressively fills the bar with color to reflect loading progress.
- **Updated `layout.tsx` to include the `LoadingOverlay` component**:
  - Added `loading` and `progress` states to control the overlay’s visibility and progress percentage.
  - Mocked loading progress with an interval function to simulate dataset loading.
- **Applied Tailwind CSS styles** to ensure the overlay appears on top of all other elements with a semi-transparent background.

---

### Testing Instructions

1. Run the development server:
   ```bash
   nx serve frontend
2. Trigger the loading overlay by calling updateWildfireLayer in the Layout component or simulate dataset loading.

3. Verify:
  -- The loading overlay appears and dims the background when loading starts.
  -- The loading bar fills up smoothly from 0% to 100%.
  -- The overlay disappears once loading completes, allowing interaction with other elements again.


---

### Additional Notes
![image](https://github.com/user-attachments/assets/69196a74-80ee-4c44-93e2-5cc6dfafcf83)
